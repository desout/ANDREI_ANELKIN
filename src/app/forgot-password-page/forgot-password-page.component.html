<div class="container">
  <ng-container *ngIf="!(userService.currentUser$ | async)?.name;then isNotLoggedTemplate; else isLoggedTemplate"></ng-container>
  <ng-template #isNotLoggedTemplate>
    <h1>{{ 'INFO.FORGOTPAGETITLE' | translate }}</h1>
    <form (ngSubmit)="onSubmit()" [formGroup]="updatePasswordForm" #formDir="ngForm">
      <div class="form-group">
        <label>{{'INFO.USERNAME' | translate}}
          <input class="form-control" formControlName="userName">
        </label>
        <div *ngIf="userName.pending">{{'ERROR.VALIDATING' | translate}}</div>
        <div *ngIf="userName.invalid && (userName.dirty || userName.touched)"
             class="alert alert-danger">
          <div *ngIf="userName.errors['required']">
            {{'INFO.USERNAME' | translate}} {{'ERROR.REQUIRED' | translate}}.
          </div>
          <div *ngIf="userName.errors['isBadName']">
            {{'ERROR.USERNOTFOUND' | translate}}
          </div>
        </div>
      </div>

      <div class="form-group">
        <label>{{'INFO.OLDPASSWORD' | translate}}
          <input type="password"  class="form-control" formControlName="oldPassword">
        </label>
        <div *ngIf="oldPassword.invalid && (oldPassword.dirty || oldPassword.touched)"
             class="alert alert-danger">
          <div *ngIf="oldPassword.errors['required']">
            {{'INFO.PASSWORD' | translate}} {{'ERROR.REQUIRED' | translate}}.
          </div>
          <div *ngIf="oldPassword.errors['minlength']">
            {{'ERROR.MINLENGTH' | translate}}} 4
          </div>
        </div>
      </div>
      <div class="form-group">
        <label>{{'INFO.NEWPASSWORD' | translate}}
          <input type="password" class="form-control" formControlName="newPassword">
        </label>
        <div *ngIf="newPassword.invalid && (newPassword.dirty || newPassword.touched)"
             class="alert alert-danger">
          <div *ngIf="newPassword.errors['required']">
            {{'INFO.PASSWORD' | translate}} {{'ERROR.REQUIRED' | translate}}.
          </div>
          <div *ngIf="newPassword.errors['minlength']">
            {{'ERROR.MINLENGTH' | translate}}} 4
          </div>
          <div *ngIf="newPassword.errors['badPassword']">
            {{'ERROR.BADPASSWORDLOGIN' | translate}}
          </div>
        </div>
      </div>
      <button type="submit" class="btn btn-default" [disabled]="updatePasswordForm.invalid">{{'INFO.SUBMITBTN' | translate}}</button>
    </form>
  </ng-template>
  <ng-template #isLoggedTemplate>
    <h1>{{ 'LOGIN.LOGGED' | translate }}</h1>
    <button (click)="onClickLogout()" class="btn btn-danger align-self-lg-center"> {{'INFO.LOGOUT' | translate}}</button>
  </ng-template>

</div>
