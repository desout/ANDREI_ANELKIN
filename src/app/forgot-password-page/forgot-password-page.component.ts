import { Component, OnInit } from '@angular/core';
import {AuthService} from '../auth.service';
import {FormControl, FormGroup, Validators} from '@angular/forms';
import {UpdatePasswordUser} from '../../../server/server/serverWorker';
import {map} from 'rxjs/operators';
import {Router} from '@angular/router';
import {UserService} from '../user.service';
import {UsersState, CurUserState} from '../store';
import {Store} from '@ngrx/store';
import {LogoutUser} from '../store/current-user-store/actions/currentUser.actions';
import {UpdatePasswordUserFn} from '../store/user-store/actions/users.actions';

@Component({
  selector: 'app-forgot-password-page',
  templateUrl: './forgot-password-page.component.html',
  styleUrls: ['./forgot-password-page.component.scss']
})
export class ForgotPasswordPageComponent implements OnInit {
  user: UpdatePasswordUser = { name: '', oldPassword: '', newPassword: ''};
  updatePasswordForm: FormGroup;
  constructor(private router: Router,
              protected userService: UserService,
              private sessionStore: Store<CurUserState>,
              private usersStore: Store<UsersState>) { }

  ngOnInit() {
    this.updatePasswordForm = new FormGroup({
      userName: new FormControl(this.user.name, {
        validators: [Validators.required]
      }),
      oldPassword: new FormControl(this.user.oldPassword, {validators: [Validators.required, Validators.minLength(4)]}),
      newPassword: new FormControl(this.user.newPassword, {validators: [Validators.required, Validators.minLength(4)]})
    });
  }
  onSubmit() {
    this.usersStore.dispatch(new UpdatePasswordUserFn({
      name: this.userName.value,
      oldPassword: this.oldPassword.value,
      newPassword: this.newPassword.value} as UpdatePasswordUser));
  }
  onClickLogout() {
    this.sessionStore.dispatch(new LogoutUser());
  }
  get userName() {
    return this.updatePasswordForm.get('userName');
  }

  get oldPassword() {
    return this.updatePasswordForm.get('oldPassword');
  }
  get newPassword() {
    return this.updatePasswordForm.get('newPassword');
  }
}
