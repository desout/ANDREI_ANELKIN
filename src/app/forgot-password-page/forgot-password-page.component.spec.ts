import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ForgotPasswordPageComponent } from './forgot-password-page.component';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {TestModule} from '../Test.module';

describe('ForgotPasswordPageComponent', () => {
  let component: ForgotPasswordPageComponent;
  let fixture: ComponentFixture<ForgotPasswordPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [],
      schemas: [NO_ERRORS_SCHEMA],
      imports: [TestModule],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ForgotPasswordPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
