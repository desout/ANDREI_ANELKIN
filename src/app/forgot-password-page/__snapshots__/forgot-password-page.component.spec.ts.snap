// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordPageComponent should create 1`] = `
<div
  id="root0"
  ng-version="7.0.3"
>
  <div
    class="container"
  >
    
    <h1>
      INFO.FORGOTPAGETITLE
    </h1>
    <form
      class="ng-untouched ng-pristine ng-invalid"
      ng-reflect-form="[object Object]"
      novalidate=""
    >
      <div
        class="form-group"
      >
        <label>
          INFO.USERNAME 
          <input
            class="form-control ng-untouched ng-pristine ng-invalid"
            formcontrolname="userName"
            ng-reflect-name="userName"
          />
        </label>
        
        
      </div>
      <div
        class="form-group"
      >
        <label>
          INFO.OLDPASSWORD 
          <input
            class="form-control ng-untouched ng-pristine ng-invalid"
            formcontrolname="oldPassword"
            ng-reflect-name="oldPassword"
            type="password"
          />
        </label>
        
      </div>
      <div
        class="form-group"
      >
        <label>
          INFO.NEWPASSWORD 
          <input
            class="form-control ng-untouched ng-pristine ng-invalid"
            formcontrolname="newPassword"
            ng-reflect-name="newPassword"
            type="password"
          />
        </label>
        
      </div>
      <button
        class="btn btn-default"
        disabled=""
        type="submit"
      >
        INFO.SUBMITBTN
      </button>
    </form>
    
    
  </div>
</div>
`;
